keras_cv/layers/feature_pyramid.py

```diff
@@ -0,0 +1,208 @@
+        self.pyramid_levels = sorted(pyramid_levels)
+        self.num_channels = num_channels
+
+        # required for successful serialization
+        self.lateral_layers_passed = lateral_layers
+        self.output_layers_passed = output_layers
+        self.top_down_op_passed = top_down_op
+        self.merge_op_passed = merge_op
+
+        if not lateral_layers:
+            # populate self.lateral_ops with default FPN Conv2D 1X1 layers
+            self.lateral_layers = {}
+            for i in self.pyramid_levels:
+                self.lateral_layers[i] = tf.keras.layers.Conv2D(
+                    self.num_channels,
+                    kernel_size=1,
+                    strides=1,
+                    padding="same",
+                    name=f"lateral_P{i}",
+                )
+
+        else:
+            if (
+                not isinstance(lateral_layers, dict)
+                or sorted(lateral_layers.keys()) != self.pyramid_levels
+            ):
+                raise ValueError(
+                    f"Expect lateral_layers to be a dict with keys as "
+                    f"{self.pyramid_levels}, got {lateral_layers}"
+                )
+            self.lateral_layers = lateral_layers
+
+        # Output conv2d layers.
+        if not output_layers:
+            self.output_layers = {}
+            for i in self.pyramid_levels:
+                self.output_layers[i] = tf.keras.layers.Conv2D(
+                    self.num_channels,
+                    kernel_size=3,
+                    strides=1,
+                    padding="same",
+                    name=f"output_P{i}",
+                )
+        else:
+            if (
+                not isinstance(output_layers, dict)
+                or sorted(output_layers.keys()) != self.pyramid_levels
+            ):
+                raise ValueError(
+                    f"Expect output_layers to be a dict with keys as "
+                    f"{self.pyramid_levels}, got {output_layers}"
+                )
+            self.output_layers = output_layers
+
+        # the same upsampling layer is used for all levels
+        if not top_down_op:
+            self.top_down_op = tf.keras.layers.UpSampling2D(size=2)
+        else:
+            self.top_down_op = top_down_op
+        # the same merge layer is used for all levels
+        if not merge_op:
+            self.merge_op = tf.keras.layers.Add()
+        else:
+            self.merge_op = merge_op
+
+    def call(self, features):
+        # Note that this assertion might not be true for all the subclasses. It is
+        # possible to have FPN that has high levels than the height of backbone outputs.
+        if (
+            not isinstance(features, dict)
+            or sorted(features.keys()) != self.pyramid_levels
+        ):
+            raise ValueError(
+                "Expect the input features to be a dict with int keys that"
+                f"matches to the {self.pyramid_levels}, got: {features}"
```

Comment:
lets tell users where the self.pyramid_levels dict that will be formatted in is coming from.

FeaturePyramid expects input features to be a dict with int keys that match the values provided in `pyramid_levels`.  want keys(inputs)={self.pyramid_levels}, got={features.keys()}